packages/flutter/test/material/slider_test.dart

```diff
@@ -2375,6 +2375,99 @@ void main() {
     expect(bottomSliderValue, 0.5, reason: 'unfocused bottom Slider unaffected by third arrowRight');
   });
 
+testWidgets('When adjusting slider with arrow keys onChangeStart and onChangeEnd callbacks are activated and display correct values', (WidgetTester tester) async
```

Comment:
You actually fixed several things in this change and are testing them all in one test.  It's generally best practice in unit testing to test one thing per test - then when the test fails you can more easily go right to the problem.  I'd consider taking each of the places where you have a cluster of asserts and make that a separate test (possibly abstracting out a helper function to reduce redundant code for setting up the tests).

This is a very light suggestion.  It looks like the Flutter team doesn't do a great job of abiding by the principal I noted.  They may prefer to make sure they have coverage, so at a quick glance I'd guess this meets Flutter's standards.

Reply:
That makes sense, I will refactor.